service/publish/handler.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"toktik/repo/model"
+
+	"github.com/bakape/thumbnailer/v2"
+	"github.com/gofrs/uuid"
+
+	"image/jpeg"
+
+	publish "toktik/kitex_gen/douyin/publish"
+	gen "toktik/repo"
+	"toktik/storage"
+)
+
+// getThumbnail Generate JPEG thumbnail from video
+func getThumbnail(input io.ReadSeeker) ([]byte, error) {
+	_, thumb, err := thumbnailer.Process(input, thumbnailer.Options{})
+	if err != nil {
+		return nil, errors.New("failed to create thumbnail")
+	}
+	buf := new(bytes.Buffer)
+	err = jpeg.Encode(buf, thumb, nil)
+	if err != nil {
+		return nil, errors.New("failed to create buffer")
+	}
+	return buf.Bytes(), nil
+}
+
+// PublishServiceImpl implements the last service interface defined in the IDL.
+type PublishServiceImpl struct{}
+
+// CreateVideo implements the PublishServiceImpl interface.
+func (s *PublishServiceImpl) CreateVideo(ctx context.Context, req *publish.CreateVideoRequest) (resp *publish.CreateVideoResponse, err error) {
+	if http.DetectContentType(req.Data) != "video/mp4" {
+		return nil, errors.New("invalid content type")
```

Comment:
Handle errors that have nothing to do with the RPC process, or improve with [Kitex Biz Exception](https://www.cloudwego.io/docs/kitex/tutorials/basic-feature/bizstatuserr/), I will integrate it in the next sprint, please feel free to start if you are intrested.

Reply:
Example for err handling from: [feats-merge-feed&publish](https://github.com/Toktik-Team/toktik/blob/a08cafc7e836c830921b7b8ed82b6c8cbf9626c7/service/feed/handler.go#L23)